hyper/common/exceptions.py

```diff
@@ -71,3 +71,10 @@ class MissingCertFile(Exception):
     The certificate file could not be found.
     """
     pass
+
+
+class ProxyError(Exception):
```

Comment:
I'm happy to introduce a common `ConnectionError` that the HTTP/2 one subclasses.